// wraptmdb-ts/publish/src/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetHeader = exports.GetURL = exports.GetToken = exports.SetHeader = exports.SetToken = exports.Route = exports.header = void 0;
/*
 * The MIT License (MIT)
 *
 * Copyright (c) kwangsing3
 *
 * https://github.com/wrapTMDB/wrapTMDB-ts
 *
 */
exports.header = {
    'User-Agent': '',
    Referer: '',
};
const Common = {
    TOKEN: '',
    BASE_URL: 'https://api.themoviedb.org/3/',
    HEADER: exports.header,
};
exports.Route = {
    MOVIE: 'movie/',
    TV: 'tv/',
    ACCOUNT: 'account/',
    CERTIFICATION: 'certification/',
    AUTHENTICATION: 'Authentication/',
    COLLECTION: 'colleciton/',
    COMPANY: 'company/',
    CONFIGURATION: 'Configuration/',
    CREDIT: 'credit/',
    DISCOVER: 'discover/',
    FIND: 'find/',
    GENRE: 'genre/',
    GUESTSESSION: 'guest_session/',
    KEYWORD: 'keyword/',
    LIST: 'list/',
    NETWORK: 'network/',
    TRENDING: 'trending/',
    PEOPLE: 'person/',
    REVIEWS: 'reviews/',
    SEARCH: 'search/',
    EPISODE: 'episode/',
    EPISODEGROUPS: 'episode_group/',
    WATCHPROVIDERS: 'watch/providers/',
};
/********************
 * Set Token
 *********************/
function SetToken(token) {
    Common.TOKEN = token;
}
exports.SetToken = SetToken;
/********************
 * Set RESTful request Header
 *********************/
function SetHeader(input) {
    if (Object.prototype.hasOwnProperty.call(input, 'User-Agent'))
        Common.HEADER['User-Agent'] = input['User-Agent'];
    if (Object.prototype.hasOwnProperty.call(input, 'Referer'))
        Common.HEADER['Referer'] = input['Referer'];
}
exports.SetHeader = SetHeader;
/********************
 * Debug function to get TOKEN
 * @returns {string} Common.TOKEN
 ********************/
function GetToken() {
    return Common.TOKEN;
}
exports.GetToken = GetToken;
/********************
 * Get Api hostname
 * @returns {string} TMDB API hostname
 ********************/
function GetURL() {
    return Common.BASE_URL;
}
exports.GetURL = GetURL;
/********************
 * Debug function to get TOKEN
 * @returns {string} Common.TOKEN
 ********************/
function GetHeader() {
    return Common.HEADER;
}
exports.GetHeader = GetHeader;
/*
  -axios-

  axios.get(url, config)
  axios.delete(url, config)
  axios.head(url, config)
  axios.post(url, data, config)
  axios.put(url, data, config)
  axios.patch(url, data, config)

*/
