// wraptmdb-ts/publish/src/people.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.People = void 0;
/*
 * The MIT License (MIT)
 *
 * Copyright (c) kwangsing3
 *
 * https://github.com/wrapTMDB/wrapTMDB-ts
 *
 */
const axios = require('axios');
const c_module = require("./common");
const baseURL = c_module.GetURL();
class People {
    /********************
     * 1.GET /person/{person_id}
     * @description Get the primary person details by id.
     * @param {number|string} person_id
     * @param {string} language
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-details
     ********************/
    async GetDetails(person_id, language, append_to_response) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL + c_module.Route.PEOPLE + `${person_id}` + `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (append_to_response !== '' && append_to_response !== undefined)
            targetURL += `&append_to_response=${append_to_response}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 2.GET /person/{person_id}/changes
     * @description Get the changes for a person. By default only the last 24 hours are returned.
     * @param {number|string} person_id
     * @param {string} end_date (optional)
     * @param {number} page (optional)
     * @param {string} start_date (optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-changes
     ********************/
    async GetChanges(person_id, start_date, end_date, page) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/changes` +
            `?api_key=${token}`;
        if (start_date !== '' && start_date !== undefined)
            targetURL += `&start_date=${start_date}`;
        if (end_date !== '' && end_date !== undefined)
            targetURL += `&end_date=${end_date}`;
        if (page !== undefined)
            targetURL += `&page=${page}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 3.GET /person/{person_id}/movie_credits
     * @description Get the movie credits for a person.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-movie-credits
     ********************/
    async GetMovieCredits(person_id, language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/movie_credits` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 4.GET /person/{person_id}/tv_credits
     * @description Get the TV show credits for a person.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-tv-credits
     ********************/
    async GetTVCredits(person_id, language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/tv_credits` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 5.GET /person/{person_id}/combined_credits
     * @description Get the movie and TV credits together in a single response.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-combined-credits
     ********************/
    async GetCombinedCredits(person_id, language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/combined_credits` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 6.GET /person/{person_id}/external_ids
     * @description Get the external ids for a person. We currently support the following external sources.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-external-ids
     ********************/
    async GetExternalIDs(person_id, language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/external_ids` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 7.GET /person/{person_id}/images
     * @description Get the images for a person.
     * @param {number|string} person_id
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-images
     ********************/
    async GetImages(person_id) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        const targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/images` +
            `?api_key=${token}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 8.GET /person/{person_id}/tagged_images
     * @description Get the primary person details by id.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @param {number} page(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-tagged-images
     ********************/
    async GetTaggedImages(person_id, language, page) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/tagged_images` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (page !== undefined) {
            targetURL += `&page=${page}`;
        }
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 9.GET /person/{person_id}/translations
     * @description Get a list of translations that have been created for a person.
     * @param {number|string} person_id
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-translations
     ********************/
    async GetTranslations(person_id, language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL +
            c_module.Route.PEOPLE +
            `${person_id}/translations` +
            `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 10.GET /person/latest
     * @description Get the most newly created person. This is a live response and will continuously change.
     * @param {string} language(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-latest-person
     ********************/
    async GetLatest(language) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL + c_module.Route.PEOPLE + 'latest' + `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
    /********************
     * 11.GET /person/{person_id}
     * @description Get the primary person details by id.
     * @param {string} language(optional)
     * @param {number} page(optional)
     * @returns {any} JSON
     * @doc https://developers.themoviedb.org/3/people/get-person-details
     ********************/
    async GetPopular(language, page) {
        const token = c_module.GetToken();
        const header = c_module.GetHeader();
        let targetURL = baseURL + c_module.Route.PEOPLE + 'popular' + `?api_key=${token}`;
        if (language !== '' && language !== undefined)
            targetURL += `&language=${language}`;
        if (page !== undefined) {
            targetURL += `&page=${page}`;
        }
        if (token === 'UnitTest_api_key') {
            return targetURL;
        }
        const data = await axios.get(targetURL, header);
        return data.data;
    }
}
exports.People = People;
/*
  1.GET Get Details
  2.GET Get Changes
  3.GET Get Movie Credits
  4.GET Get TV Credits
  5.GET Get Combined Credits
  6.GET Get External IDs
  7.GET Get Images
  8.GET Get Tagged Images
  9.GET Get Translations
  10.GET Get Latest
  11.GET Get Popular
*/
