// wraptmdb-ts/publish/src/tvepisodes.js
'use strict';
Object.defineProperty(exports, '__esModule', {value: true});
exports.TVEpisodes = void 0;
/*
 * The MIT License (MIT)
 *
 * Copyright (c) kwangsing3
 *
 * https://github.com/wrapTMDB/wrapTMDB-ts
 *
 */
const axios = require('axios');
const c_module = require('./common');
const baseURL = c_module.GetURL();
class TVEpisodes {
  /********************
   * 1.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}
   * @description Get the TV season details by id.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {string} language (optional)
   * @param {string} append_to_response (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-details
   ********************/
  async GetDetails(
    tv_id,
    season_number,
    episode_number,
    language,
    append_to_response
  ) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      `?api_key=${token}`;
    if (language !== '' && language !== undefined)
      targetURL += `&language=${language}`;
    if (append_to_response !== '' && append_to_response !== undefined)
      targetURL += `&append_to_response=${append_to_response}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 2.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/account_states
   * @description Get your rating for a episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {string} guest_session_id (optional)
   * @param {string} session_id (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-account-states
   ********************/
  async GetAccountStates(
    tv_id,
    season_number,
    episode_number,
    guest_session_id,
    session_id
  ) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}/` +
      'account_states' +
      `?api_key=${token}`;
    if (guest_session_id !== '' && guest_session_id !== undefined)
      targetURL += `&guest_session_id=${guest_session_id}`;
    if (session_id !== '' && session_id !== undefined)
      targetURL += `&session_id=${session_id}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 3.GET /tv/episode/{episode_id}/changes
   * @description Get the changes for a TV season. By default only the last 24 hours are returned.
   * @param {number|string} episode_id
   * @param {string} start_date (optional)
   * @param {string} end_date (optional)
   * @param {number} page (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-changes
   ********************/
  async GetChanges(episode_id, start_date, end_date, page) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      c_module.Route.EPISODE +
      `${episode_id}` +
      '/changes' +
      `?api_key=${token}`;
    if (start_date !== '' && start_date !== undefined)
      targetURL += `&start_date=${start_date}`;
    if (end_date !== '' && end_date !== undefined)
      targetURL += `&end_date=${end_date}`;
    if (page !== undefined) {
      targetURL += `&page=${page}`;
    }
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 4.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/credits
   * @description Get the credits (cast, crew and guest stars) for a TV episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {string} language (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-credits
   ********************/
  async GetCredits(tv_id, season_number, episode_number, language) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/credits' +
      `?api_key=${token}`;
    if (language !== '' && language !== undefined)
      targetURL += `&language=${language}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 5.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/external_ids
   * @description Get the TV season details by id.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-external-ids
   ********************/
  async GetExternalIDs(tv_id, season_number, episode_number) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    const targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/external_ids' +
      `?api_key=${token}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 6.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/images
   * @description Get the images that belong to a TV episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-images
   ********************/
  async GetImages(tv_id, season_number, episode_number) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    const targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/images' +
      `?api_key=${token}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 7.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/translations
   * @description Get the translation data for an episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-translations
   ********************/
  async GetTranslations(tv_id, season_number, episode_number) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    const targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/translations' +
      `?api_key=${token}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
  /********************
   * 8.POST /tv/{tv_id}/season/{season_number}/episode/{episode_number}/rating
   * @description Rate a TV episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {any} query
   * @param {string} session_id (optional)
   * @param {string} guest_session_id (optional)
   * @returns JSON
   * @example query{
   *    "value": 8.5
   * }
   * @doc https://developers.themoviedb.org/3/tv-episodes/rate-tv-episode
   ********************/
  async PostRateTVEpisode(
    tv_id,
    season_number,
    episode_number,
    query,
    session_id,
    guest_session_id
  ) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/rating' +
      `?api_key=${token}`;
    if (guest_session_id !== '' && guest_session_id !== undefined)
      targetURL += `&guest_session_id=${guest_session_id}`;
    if (session_id !== '' && session_id !== undefined)
      targetURL += `&session_id=${session_id}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.post(targetURL, query, header);
    return data.data;
  }
  /********************
   * 9.DELETE /tv/{tv_id}/season/{season_number}/episode/{episode_number}/rating
   * @description Remove your rating for a TV episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {string} session_id (optional)
   * @param {string} guest_session_id (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/delete-tv-episode-rating
   ********************/
  async DeleteDeleteRating(
    tv_id,
    season_number,
    episode_number,
    session_id,
    guest_session_id
  ) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/rating' +
      `?api_key=${token}`;
    if (guest_session_id !== '' && guest_session_id !== undefined)
      targetURL += `&guest_session_id=${guest_session_id}`;
    if (session_id !== '' && session_id !== undefined)
      targetURL += `&session_id=${session_id}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.delete(targetURL, {}, header);
    return data.data;
  }
  /********************
   * 10.GET /tv/{tv_id}/season/{season_number}/episode/{episode_number}/videos
   * @description Get the videos that have been added to a TV episode.
   * @param {number|string} tv_id
   * @param {number|string} season_number
   * @param {number|string} episode_number
   * @param {string} language (optional)
   * @returns JSON
   * @doc https://developers.themoviedb.org/3/tv-episodes/get-tv-episode-videos
   ********************/
  async GetVideos(tv_id, season_number, episode_number, language) {
    const token = c_module.GetToken();
    const header = c_module.GetHeader();
    let targetURL =
      baseURL +
      c_module.Route.TV +
      `${tv_id}/` +
      'season/' +
      `${season_number}/` +
      c_module.Route.EPISODE +
      `${episode_number}` +
      '/videos' +
      `?api_key=${token}`;
    if (language !== '' && language !== undefined)
      targetURL += `&language=${language}`;
    if (token === 'UnitTest_api_key') {
      return targetURL;
    }
    const data = await axios.get(targetURL, header);
    return data.data;
  }
}
exports.TVEpisodes = TVEpisodes;
/*
  1.GET Get Details
  2.GET Get Account States
  3.GET Get Changes
  4.GET Get Credits
  5.GET Get External IDs
  6.GET Get Images
  7.GET Get Translations
  8.POST Rate TV Episode
  9.DELETE Delete Rating
  10.GET Get Videos
*/
